### DISCUSSION TITLE
Поддержка шрифта...

### DISCUSSION THREAD

**Discussion by CJRomix** [Kolcha/DigitalClock5#4] [General]
**Привет и большое спасибо за программу!** Давно использую версию 4 и недавно решил попробовать новую 5 версию... Всё вроде бы хорошо работает. Но вот я использую очень хороший шрифт Marske и вот как у меня это выглядит: [2024-09-18_123903](https://github.com/user-attachments/assets/ae8b78ca-d68c-4079-bbf8-02ff00405136) в основном хочется что бы появилась регулировка расстояния между строк, та что имеется не регулирует почему-то, забыл сказать что это в основном касается Плагина "Дата".  **Спасибо!**
P.S. Шрифт приложу так же. 
[Marske.zip](https://github.com/user-attachments/files/17042011/Marske.zip)

**Comment by Kolcha** [comment]
для работы со шрифтами я бы и рекомендовал только 5-ю версию, в 4-й поддержка шрифтов вся на хаках (но с моноширинными должно быть хорошо и незаметно).

да, пока возможности менять расстояние между строк для плагнов невозможно (хотя программная возможность есть, просто недоступно из интерфейса). это будет добавлено в ближайшее время, т.к. не требует больших изменений.

возможно я пересмотру текущее решение и сделаю так чтобы имеющиеся контроллы также влияли и на плагины.

то что оно выглядит так плотно говорит о том что шрифт содержит реально минимально возможные расстояния/размеры символов, без какого-либо дополнительного места вокруг (особенно по вертикали), что крайне большая редкость. я даже планирую добавить опцию чтобы игнорировать вертикальные расстояния между символами и использовать свои вместо того что написано в шрифте, ибо обычно они уж очень большие (особенно для разных вычурных шрифтов) и это выглядит некрасиво, но тут ситуация обратная...

в любом случае, со следующим обновлением решение будет. правда обновление не раньше чем через 2 недели, где-то к концу месяца, сейчас мне не до того, увы. и так эти 2 "багфикс" версии в спешке делал только из-за критичных фиксов (ну и с ними еще то что юзеры просили и было просто добавить)...

**Reply by CJRomix** [reply]
Спасибо, будем ждать, торопиться некуда ;)

**Comment by CJRomix** [comment]
Еще мне очень зашла фишка с часами в трее, прям респект! Но хотелось бы чтоб они не синими были, а белыми или этот цвет можно было бы выбирать, думаю чтоб не усложнять можно ограничиться выбором под тему интерфейса Windows (белый/черный).
Еще раз Спасибо и Удачи!

**Comment by Kolcha** [comment]
фишка с часами в трее была написана "от нечего делать" и забавы ради (хотел кое-чего попробовать) просто как "странная ненужная фича", но что-то нетипичное и потенциально интересное.

по поводу цвета под тему у меня уже давно пунктик есть, но на данный момент есть куда более насущные вопросы. к тому же детект темы явно нетривиальная задача, тем более для меня, ибо я даже не пользователь винды, не то что разработчик под нее. я очень далек от виндовых технологий, и уже давным давно забыл и то что знал ибо не востребовано было уже более 10 лет, хотя и начинал с винды когда-то... а то что многое попросту изменилось и говорить нечего...

но это как-нибудь сделаю, ибо вероятно кто-нибудь тоже попросит. и вообще я планирую перетащить из старых часов логику позволяющую перекрашивать любые значки в программе в зависимости от темы (но этот трей - отдельная история, там оно в прямом смысле на ходу рисует, там нет картинки), в свое время написанную сугубо для линукса, но потом пригодилось и на маке, а теперь и на винде можно заюзать.

а текущий цвет там захардкодан из темы значков что я использую в разных элементах интерфейса, просто цвет мне понравился, а главное хорошо контрастирует как со светлым так и темным фоном.

**Reply by CJRomix** [reply]
Вчера установил новую Винду 11 (24H2), вот так у меня выглядят часики и трей:
![Desk_W11_24H2](https://github.com/user-attachments/assets/40535b31-142e-4df1-964e-590c18c2bd82)

**Reply by Kolcha** [reply]
да уж, с таким фоном часы в трее едва заметны...

**Comment by Kolcha** [comment]
вообщем, хорошие новости - я зарелизил версию 5.0.3, которая включает все пожелания:

- изменение любых расстояний (включая межстрочный интервал для плагинов)
- часы в трее меняют цвет в зависимости от темы (оказалось мой старый код ещё работает)

так что можно закрывать этот "тикет", если всё устраивает

лучше при необходимости открыть другой (и как issue) если есть конкретные замечания или предложения, так проще отслеживать

**Reply by CJRomix** [reply]
Огромный Респект! Теперь всё тип-топчик ;)
